int16 id
float32 X
float32 Y
int16 value
int16 patch
int16 flag
